File: php/features.php
<?php
declare(strict_types=1);

// FlyffGame SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FlyffGameFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FlyffGameBaseFeature();
            case "test":
                return new FlyffGameTestFeature();
            default:
                return new FlyffGameBaseFeature();
        }
    }
}

File: py/features.py
# FlyffGame SDK feature factory

from feature.base_feature import FlyffGameBaseFeature
from feature.test_feature import FlyffGameTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FlyffGameBaseFeature(),
        "test": lambda: FlyffGameTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
